service/history/decision/handler.go

```diff
@@ -725,7 +725,15 @@ func (handler *handlerImpl) handleBufferedQueries(
 
 	// Consistent query requires both server and client worker support. If a consistent query was requested (meaning there are
 	// buffered queries) but worker does not support consistent query then all buffered queries should be failed.
-	if versionErr := handler.versionChecker.SupportsConsistentQuery(clientImpl, clientFeatureVersion); versionErr != nil {
+	versionErr := handler.versionChecker.SupportsConsistentQuery(clientImpl, clientFeatureVersion)
+	// todo (David.Porter) remove the skip on version check for
+	// clientImpl and clientFeatureVersion where they're nil
+	// There's a bug, probably in matching somewhere which isn't
+	// forwarding the client headers for version
+	// info correctly making this call erroneously fail sometimes.
+	// https://t3.uberinternal.com/browse/CDNC-8641
+	// So defaulting just this flow to fail-open in the absence of headers.
+	if versionErr != nil && clientImpl != "" && clientFeatureVersion != "" {
```

Comment:
Hmm sure, this is fine for now

Reply:
there are code paths within vesionChecker that handles `clientImpl == nil` gracefully. so expecting both `clientImpl` and `clientFeatureVersion` to be empty might exclude some genuine cases.

Reply:
discussed offline, I believe we concluded that the clients that didn't support this and were using QueryConsistencyStrong to be a small set.